### PR TITLE
Add better ipytree error handling in jupyter notebooks

### PR DESCRIPTION
**Pull request by mzjp2**
Resolves https://github.com/zarr-developers/zarr-python/issues/537

Adds a custom error message when failing to import `ipytree` in jupyter notebooks along with instructions on how to get it (`pip install zarr[jupyter]`). Also added `jupyter` itself to `extras_require["jupyter"]` seeing as `jupyter` isn't a requirement in-and-of-itself of `zarr`. Let me know if anything doesn't make sense, happy to change.

TODO:
* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [x] AppVeyor and Travis CI passes
* [x] Test coverage is 100% (Coveralls passes)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -24,14 +24,15 @@
...
+        'write_to': 'zarr/version.py',
     },
     setup_requires=[
         'setuptools>=38.6.0',
-        'setuptools-scm>1.5.4'
+        'setuptools-scm>1.5.4',
     ],
     extras_require={
         'jupyter': [
+            'jupyter',
```
Comment (jakirkham) [inline setup.py]: Should we just do `notebook`? This seems to install `qtconsole` and other things users may not need/want.

ref: https://pypi.org/project/jupyter/#description

Comment (mzjp2) [reply setup.py]: Good point, I didn't realise `pip install notebook` was a thing. Changing it now. :)

**Comment by mzjp2** [comment]
Maybe there's a place in our documentation we can put it in as a note? I'll have a hunt and see if it makes sense anywhere.

Also, no worries re: turnaround time, thank you for having a look. :)

BTW, do you know what's up with the CI?

**Comment by jakirkham** [comment]
Yeah that is a good question 😅

Travis CI is pushing everything to `.com` instead of `.org`. That appears to be sort of working. Remaining issues:

1. Update protected branch (`.com` not listed here?)
2. Fix Coveralls (not sure why it doesn't respect `.com`)

Thoughts? 🙂

**Comment by mzjp2** [comment]
Oh, I suppose that line can also go in the error message. Which do you prefer?

**Comment by jakirkham** [comment]
Adding it to the error message sounds good. Probably somewhere in the docs is also good.

FWIW I added [a line at the bottom of this section]( https://zarr.readthedocs.io/en/stable/tutorial.html#array-and-group-diagnostics ), but it sounds like that is not the best place for it.

**Comment by joshmoore** [comment]
Closing and re-opening to see if coveralls will go green.

**Comment by mzjp2** [comment]
Happy to merge @jakirkham?

File: zarr/util.py [hunk 2]
```diff
@@ -428,7 +428,14 @@ def tree_widget_sublist(node, root=False, expand=False):
 
 
 def tree_widget(group, expand, level):
-    import ipytree
+    try:
+        import ipytree
+    except ImportError as error:
+        raise ImportError(
+            "{}: Run `pip install zarr[jupyter]` to get the required ipytree "
```
Comment (jakirkham) [inline zarr/util.py:435]: What about "Install `ipytree` with `conda` or `pip` ..."?

**Comment by jakirkham** [comment]
Had one suggestion about the error message. Otherwise agree this looks good.

File: zarr/util.py (hunk 2 above)
Comment (mzjp2) [reply zarr/util.py:435]: Could do, but seems weird to tell users to install a dependency if it wasn't explicitly stated as one of our dependencies, whereas `pip install zarr[jupyter]` shows that it's an _optional_ dependency of ours. I'm not fussed and happy to change it if you'd prefer though. 😄

Comment (jakirkham) [reply zarr/util.py:435]: Yeah the main thing I'm trying to address is to make sure that Conda users handle the install through Conda (bracket notation is not a thing for Conda) and Pip wouldn't use Conda.

Not attached to the wording if you have other thoughts on how we might clarify this optional dependency to users.

Comment (mzjp2) [reply zarr/util.py:435]: Ah good point, I didn't think of `conda`. I'll include that in the error message. :)

Comment (jakirkham) [reply zarr/util.py]: Looks good. Thanks! 😄

**Comment by jakirkham** [comment]
Could you please resolve the merge conflicts? Should be good to go after that 🙂

**Comment by jakirkham** [comment]
Thanks @mzjp2! 😄

**Pull request merged**